crates/kitsune_p2p/types/src/tx_utils/proxy_url.rs

```diff
@@ -138,21 +140,27 @@ impl ProxyUrl {
     }
 
     /// Extract the cert digest from the url
-    pub fn digest(&self) -> CertDigest {
+    pub fn digest(&self) -> KitsuneResult<CertDigest> {
         let scheme = self.full.scheme();
         if scheme == "wss" || scheme == "ws" {
             // override for tx5
             if let Some(mut i) = self.full.path_segments() {
                 if let Some(u) = i.next() {
-                    let digest = base64::prelude::BASE64_URL_SAFE_NO_PAD.decode(u).unwrap();
-                    return CertDigest::from_slice(&digest);
+                    let digest = base64::prelude::BASE64_URL_SAFE_NO_PAD
+                        .decode(u)
+                        .map_err(KitsuneError::other)?;
+                    return Ok(CertDigest::from_slice(&digest));
```

Comment:
Would be nice to delete all the non-tx5 processing logic from this file, but this PR is big enough already, we can do that as a follow-on.